File: frontend/src/microservices/hotel/objects/HotelBookingPlannerResponse.ts
export type HotelBookingPlannerResponse = {
  orderId: string
  orderItemId: string
  status: string
  totalPriceAmount: string
  currency: string
}

export const hotelBookingPlannerResponseFromJson = (json: string): HotelBookingPlannerResponse =>
  JSON.parse(json) as HotelBookingPlannerResponse

export const hotelBookingPlannerResponseToJson = (value: HotelBookingPlannerResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/hotel/objects/HotelBookingPlannerResponse.scala
package com.typesafe.travel.hotel.objects

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class HotelBookingPlannerResponse(
    orderId: String,
    orderItemId: String,
    status: String,
    totalPriceAmount: String,
    currency: String
)

object HotelBookingPlannerResponse:
  given sourceEncoder: Encoder[HotelBookingPlannerResponse] = deriveEncoder
  given sourceDecoder: Decoder[HotelBookingPlannerResponse] = deriveDecoder
